functions.R

```diff
@@ -59,6 +59,46 @@ get_survey_data <- function() {
         colnames(temp_dat) <- attr(temp_dat,"variable.labels")
         # get the column names we want from are varibale list
         temp_sub <- clean_subset_survey(temp_dat, get_year = get_year, folder = temp_folder)
+        
+        colnames(temp_sub)[grepl('age|Age', colnames(temp_sub))]
+        summary(as.factor(temp_sub$respondents_age))
+        # remove age groups that are above 29 
+        if(grepl('gss_2010_1|gss_2012_1', temp_data)) {
+          
+          temp_sub <- temp_sub[!grepl('30 to 34|35 to 39|40 to 44|45 to 49|50 to 54|55 to 59|60 to 64|65 to 69|70 to 74|75 to 79|80 years+', 
+                                      temp_sub$age_group_of_the_respondent_groups_of_5),]
+          
+        } else if(grepl('gss_2010_2', temp_data)) {
+          temp_sub <- temp_sub[!grepl('30 to 34|35 to 39|40 to 44|45 to 49|50 to 54|55 to 59|60 to 64|65 to 69|70 to 74|75 to 79|80 years+', 
+                                      temp_sub$age_group_of_the_respondent),]
+          
+        } else if(grepl('gss_2011', temp_data)) {
+          # note that age here should be numeric and is not grouped. 
+          # also the age at second interview seems to be corrupted, so will only use first one.
+          temp_sub$rs_age_at_time_of_survey_interview_1 <- 
+            as.numeric(as.character(temp_sub$rs_age_at_time_of_survey_interview_1))
+          # remove any over 29 - we can group these later
+          temp_sub <- temp_sub[temp_sub$rs_age_at_time_of_survey_interview_1 < 30,]
+          
+        } else if(grepl('piaac', temp_data)) {
+          temp_sub <- temp_sub[!grepl('35-44|45-54|55 plus|>65', 
```

Comment:
@benmbrew Why are you filtering for what the age group should NOT be, rather than what the age group should be? Ie, instead of saying `!grepl('a|b|c|d|e', age)`, why not just say `grepl('f', age)`?

Reply:
Because for the first few there were a lot more older ages, but yeah I'll go through and change that.

Reply:
Actually wait, that's not even true. should have done it your way from beginning. Changing now